tests/cpp/testobject/PdxType.cpp

```diff
@@ -173,13 +173,12 @@ void PdxTests::PdxType::toData(PdxWriter& pw) const {
 }
 
 void PdxTests::PdxType::fromData(PdxReader& pr) {
-  // TODO:temp added, delete later
-
   int32_t* Lengtharr;
-  _GEODE_NEW(Lengtharr, int32_t[2]);
   int32_t arrLen = 0;
+  deleteByteByteArray();
   m_byteByteArray =
       pr.readArrayOfByteArrays("m_byteByteArray", arrLen, &Lengtharr);
+  _GEODE_SAFE_DELETE_ARRAY(Lengtharr);
```

Comment:
Can we just put this is a `std::unique_ptr` and avoid the macros. This pattern is in a few files.

Reply:
I do not think that is possible in this case as the memory is reserved by PdxReader::readArrayOfByteArrays.
In order to use smart pointers here I would need to change the signature of PdxReader::readArrayOfByteArrays.